---
title: "16S_dada2_homogeneity analysis"
output: html_document
---

```{r Clear_variables}
rm(list=ls())    # Clear variables; clear plots
```

```{r Load_packages_and_workspace}
sapply(c('tidyverse','readxl','dada2', 'rmarkdown', 'knitr', 'Rcpp', 'RColorBrewer'), require, character.only = TRUE) # Load packages
theme_set(theme_bw() + theme(plot.title = element_text(size = 11, face = 'bold', hjust = 0.5)))
sessionInfo() 
WD <- getwd()
print(WD)
load(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

Deinococcus 16S copies = 3 per genome; can have up to 10 genome copies/cell (30?)
Brenneria (Dickeya) 16S copies = 7 copies per genome
Delftia acidovorans = 5 copies per genome


```{r CUTADAPT_16S_primer_trimming}
#T0 homogeneity data sets starts clock for stability assays; all samples run on a single MiSeq sequencing run completed on 06/12/23
# V4reads -- CUTADAPT

#FWD: GTGYCAGCMGCCGCGGTAA
#REV: G/GACTACNVGGGTWTCTAAT

source(paste0(dirname(dirname(WD)), "/Resources/cutadapt_v4.R") )

forward.reads <- sort(list.files(paste0(WD, "/raw_reads"), pattern = "_R1_001.fastq.gz", full.names = TRUE)) 

reverse.reads <- sort(list.files(paste0(WD, "/raw_reads"), pattern = "_R2_001.fastq.gz", full.names = TRUE)) 

sample.names <- sapply(strsplit(basename(forward.reads), "_"), `[`, 1)

cutadapt("/usr/bin/cutadapt",
         output.directory = paste0(WD, "/primertrim_reads"),
         forward.primer = "^GTGYCAGCMGCCGCGGTAA", 
         reverse.primer = "^GACTACNVGGGTWTCTAAT", 
         error.rate = 0.15,
         minumum.overlap = 19,
         clip.from.start.read1 = 0,
         clip.from.start.read2 = 1,
         maximum.N = 0,
         discard.untrimmed = TRUE,
         filter.pairs = "any",
         log.files = TRUE,
         R.cores = 4 )
```

```{r plot_quality_profile}
forward.reads <- sort(list.files(paste0(WD, "/primertrim_reads"), pattern = "_R1_cutadapt.fastq.gz", full.names = TRUE)) 
reverse.reads <- sort(list.files(paste0(WD, "/primertrim_reads"), pattern = "_R2_cutadapt.fastq.gz", full.names = TRUE)) 

# Quality Profiles of samples
plotQualityProfile(forward.reads[1:10])          
plotQualityProfile(reverse.reads[1:10])

save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_fliterAndTrim}

filtFs <- file.path(paste0(WD, "/Qtrim_reads"), paste0(sample.names, "_R1_filtered.fastq.gz"))
filtRs <- file.path(paste0(WD, "/Qtrim_reads"), paste0(sample.names, "_R2_filtered.fastq.gz"))

out <- filterAndTrim(forward.reads, filtFs, reverse.reads, filtRs, 
                     truncLen = c(200, 150),
                     maxN = c(0, 0), 
                     maxEE = c(5, 5), 
                     truncQ = c(2,2),
                     rm.phix = TRUE,
                     compress = TRUE, 
                     multithread = FALSE,
                     verbose = TRUE)

save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))

out %>% as.data.frame() %>%
  dplyr::mutate(percent.surviving = round(reads.out/reads.in*100, 1))

```

```{r DADA2_learnError_Fowardreads}
errF <- learnErrors(filtFs, multithread = TRUE, randomize = TRUE, nbases = 2e8, verbose = TRUE)
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_learnError_Reversereads}
errR <- learnErrors(filtRs, multithread = TRUE, randomize = TRUE, nbases = 2e8, verbose = TRUE)
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_plotErrors}
plotErrors(errR, nominalQ = TRUE)
plotErrors(errF, nominalQ = TRUE)
```


```{r DADA2_derepFastq}
#dropped sample V11, no reads
filtFs <- sort(list.files(paste0(WD, "/Qtrim_reads"), pattern = "_R1_filtered.fastq.gz", full.names = TRUE))
filtRs <- sort(list.files(paste0(WD, "/Qtrim_reads"), pattern = "_R2_filtered.fastq.gz", full.names = TRUE))

sample.names <- sapply(strsplit(basename(filtFs), "_"), `[`, 1)

derepFs <- derepFastq(filtFs, verbose = T)
names(derepFs) <- sample.names
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))

derepRs <- derepFastq(filtRs, verbose = T)
names(derepRs) <- sample.names
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_dada_sequence_error_correction}
dadaFs <- dada(derepFs, err = errF, multithread = TRUE, verbose = T)
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))

dadaRs <- dada(derepRs, err = errR, multithread = TRUE, verbose = T)
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_mergePairs}
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs, verbose = TRUE) 
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r DADA2_makeSequenceTable_removeBimera}
seqtab <- makeSequenceTable(mergers)
seqtab.nochim <- removeBimeraDenovo(seqtab, method = "consensus", multithread = TRUE, verbose = TRUE) 
dim(seqtab.nochim)
round(sum(seqtab.nochim)/sum(seqtab)*100, 2)
table(nchar(getSequences(seqtab)))

saveRDS(seqtab, paste0(WD, "/dada2_output/seqtab.rds"))
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r make_table_tracking_reads}
getN <- function(x) sum(getUniques(x))
if(nrow(filter(as.tibble(out), reads.in>0 & reads.out>0)) != nrow(as.tibble(out))) print("Some files didn't have reads.in or reads.out and have been dropped.")
track <- cbind(filter(as.tibble(out), reads.in>0 & reads.out>0), sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchimeric")
track

track %>%
  rownames_to_column("ID") %>%
  write.table(., paste0(WD, "/dada2_output/track.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
```

```{r DADA2_asvtab}
asvtab <- t(as.data.frame(seqtab.nochim)) %>%
  as.data.frame() %>%
  tibble::rownames_to_column() %>%
  dplyr::mutate(asvID = paste0("asv", 1:nrow(.))) %>%
  plyr::rename(c("rowname" = "Sequence"))

saveRDS(asvtab, paste0(WD, "/dada2_output/asv_seqs.rds"))

seqinr::write.fasta(as.list(asvtab$Sequence), paste0("asv", seq(nrow(asvtab))), 
                    paste0(WD, "/dada2_output/asv_seqs.fasta"))

save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r Taxonomy_assignment}
taxa <- assignTaxonomy(seqtab.nochim, paste0(dirname(dirname(WD)), "/Resources/silva_nr99_v138_wSpecies_train_set.fa.gz"), 
                       multithread=TRUE, tryRC = TRUE) %>%
  as.data.frame() %>%
  mutate(Sequence = rownames(.)) 
  
Taxa.tab <- full_join(taxa, asvtab, by = "Sequence") %>%
  gather(value = Count, key = Sample, -asvID, -Kingdom, -Phylum, -Class, -Order, -Family, -Genus, -Species, -Sequence) %>%
  group_by(Sample) %>%
  mutate(RelAbund = Count/sum(Count))

write.table(Taxa.tab, paste0(dirname(dirname(WD)), "/Processed Data/Homogeneity/taxa_tab.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))

```

```{r reformat_taxonomic_table}
taxa.tab2 <- Taxa.tab %>%
  select(-Kingdom, -Class, -Order, -Family, -Sequence, -Count) %>%
  spread(value = "RelAbund", key = "Sample")


write.table(taxa.tab2, paste0(dirname(dirname(WD)), "/Processed Data/Homogeneity/taxa_tab2.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

meta <- read_excel(paste0(WD, "/Meta_dataH.xlsx")) %>%
  rename("Sample" = "Sample name")

Taxa.tab.meta <- Taxa.tab %>%
  full_join(meta, by = "Sample")

write.table(Taxa.tab.meta, paste0(dirname(dirname(WD)), "/Processed Data/Homogeneity/Taxa.tab.meta.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

```{r uniqueASV_persample}
#omnivore
asv_unique_O <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "Omnivore" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")
  
counts_unique_O <- asv_unique_O %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

Summary_Genus_O <- Taxa.tab.meta %>%
  select(Sample, RelAbund, Group, Genus) %>%
  filter(Group == "Omnivore" & RelAbund > 0) %>%
  group_by(Sample, Genus) %>%
  mutate(RelAbund_G = sum(RelAbund)) %>%
  ungroup() %>%
  group_by(Genus) %>%
  mutate(AvgRelAbund_G = mean(RelAbund_G), MinRelAbund_G = min(RelAbund_G), MaxRelAbund_G= max(RelAbund_G)) %>%
  select(Genus, AvgRelAbund_G, MinRelAbund_G, MaxRelAbund_G) %>%
  distinct(Genus, .keep_all = T)

Summary_Asv_O <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, RelAbund, Group, Genus) %>%
  filter(Group == "Omnivore" & RelAbund > 0) %>%
  group_by(asvID) %>%
  mutate(AvgRelAbund = mean(RelAbund), MinRelAbund = min(RelAbund), MaxRelAbund= max(RelAbund)) %>%
  select(Sequence, asvID, Genus, AvgRelAbund, MinRelAbund, MaxRelAbund) %>%
  distinct(asvID, .keep_all = T)

write.table(counts_unique_O, paste0(WD, "/dada2_output/counts_unique_O.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Asv_O, paste0(WD, "/dada2_output/Summary_Asv_O.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Genus_O, paste0(WD, "/dada2_output/Summary_Genus_O.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

#vegetarian
asv_unique_V <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "Vegetarian" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")

counts_unique_V <- asv_unique_V %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

Summary_Asv_V <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, RelAbund, Group, Genus) %>%
  filter(Group == "Vegetarian" & RelAbund > 0) %>%
  group_by(asvID) %>%
  mutate(AvgRelAbund = mean(RelAbund), MinRelAbund = min(RelAbund), MaxRelAbund = max(RelAbund)) %>%
  select(asvID, Sequence, Genus, AvgRelAbund, MinRelAbund, MaxRelAbund) %>%
  distinct(asvID, .keep_all = T)

Summary_Genus_V <- Taxa.tab.meta %>%
  select(Sample, RelAbund, Group, Genus) %>%
  filter(Group == "Vegetarian" & RelAbund > 0) %>%
  group_by(Sample, Genus) %>%
  mutate(RelAbund_G = sum(RelAbund)) %>%
  ungroup() %>%
  group_by(Genus) %>%
  mutate(AvgRelAbund_G = mean(RelAbund_G), MinRelAbund_G = min(RelAbund_G), MaxRelAbund_G= max(RelAbund_G)) %>%
  select(Genus, AvgRelAbund_G, MinRelAbund_G, MaxRelAbund_G) %>%
  distinct(Genus, .keep_all = T)

write.table(counts_unique_V, paste0(WD, "/dada2_output/counts_unique_V.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Asv_V, paste0(WD, "/dada2_output/Summary_Asv_V.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Genus_V, paste0(WD, "/dada2_output/Summary_Genus_V.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

Summary_Genus <- full_join(Summary_Genus_O, Summary_Genus_V, by = "Genus") %>%
  rename(OmniAvgRelAbund_G = AvgRelAbund_G.x, OmniMinRelAbund_G = MinRelAbund_G.x, OmniMaxRelAbund_G = MaxRelAbund_G.x, 
         VegAvgRelAbund_G = AvgRelAbund_G.y, VegMinRelAbund_G = MinRelAbund_G.y, VegMaxRelAbund_G = MaxRelAbund_G.y) %>%
  replace_na(list(Genus = 'Unclassified')) %>%
  replace(is.na(.), 0)

write.table(Summary_Genus, paste0(WD, "/dada2_output/Summary_Genus.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

#negative control
asv_unique_Neg <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "water" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")

counts_unique_Neg <- asv_unique_Neg %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

#omniPool
asv_unique_OP <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "OmniPool" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")

counts_unique_OP <- asv_unique_OP %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

Summary_Asv_OP <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, RelAbund, Group, Genus) %>%
  filter(Group == "OmniPool" & RelAbund > 0) %>%
  group_by(asvID) %>%
  mutate(AvgRelAbund = mean(RelAbund), MinRelAbund = min(RelAbund), MaxRelAbund = max(RelAbund)) %>%
  select(asvID, Sequence, Genus, AvgRelAbund, MinRelAbund, MaxRelAbund) %>%
  distinct(asvID, .keep_all = T)

write.table(counts_unique_OP, paste0(WD, "/dada2_output/counts_unique_OP.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Asv_OP, paste0(WD, "/dada2_output/Summary_Asv_OP.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

#vegPool
asv_unique_VP <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "VegPool" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")

counts_unique_VP <- asv_unique_VP %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

Summary_Asv_VP <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, RelAbund, Group, Genus) %>%
  filter(Group == "VegPool" & RelAbund > 0) %>%
  group_by(asvID) %>%
  mutate(AvgRelAbund = mean(RelAbund), MinRelAbund = min(RelAbund), MaxRelAbund = max(RelAbund)) %>%
  select(asvID, Genus, Sequence, AvgRelAbund, MinRelAbund, MaxRelAbund) %>%
  distinct(asvID, .keep_all = T)

write.table(counts_unique_VP, paste0(WD, "/dada2_output/counts_unique_VP.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Asv_VP, paste0(WD, "/dada2_output/Summary_Asv_VP.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")

#zymoControl
asv_unique_Zymo <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, Count, Group) %>%
  filter(Group == "zymo" & Count > 0) %>%
  select(-Group) %>%
  spread(value = "Count", key = "Sample")

counts_unique_Zymo <- asv_unique_Zymo %>%
  summarise_all(~sum(!is.na(.))) %>%
  gather(Column, NonNA_Count)

Summary_Asv_Zymo <- Taxa.tab.meta %>%
  select(Sample, Sequence, asvID, RelAbund, Group, Genus) %>%
  filter(Group == "zymo" & RelAbund > 0) %>%
  group_by(asvID) %>%
  mutate(AvgRelAbund = mean(RelAbund), MinRelAbund = min(RelAbund), MaxRelAbund = max(RelAbund)) %>%
  select(asvID, Sequence, Genus, AvgRelAbund, MinRelAbund, MaxRelAbund) %>%
  distinct(asvID, .keep_all = T)

write.table(counts_unique_Zymo, paste0(WD, "/dada2_output/counts_unique_Zymo.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")
write.table(Summary_Asv_Zymo, paste0(WD, "/dada2_output/Summary_Asv_Zymo.txt"),
            quote = FALSE, row.names = FALSE, sep = "\t")


print(counts_unique_O)
print(Summary_Asv_O)
print(Summary_Genus_O)
print(counts_unique_V)
print(Summary_Asv_V)
print(Summary_Genus_V)
print(Summary_Genus)
print(counts_unique_Neg)
print(counts_unique_OP)
print(Summary_Asv_OP)
print(counts_unique_VP)
print(Summary_Asv_VP)
print(Summary_Asv_Zymo)
print(counts_unique_Zymo)
```


```{r}
H_level_order <- c("O1", "O2", "O3", "O4", "O5", "O6", "O7", "O8", "O9", "O10",
                   "O11", "O12", "O13", "O14", "O15", "O16", "O17", "O18", "O19", "O20",
                   "O21", "O22", "O23", "O24", "O25", "O26", "O27", "O28", "O29", "O30",
                   "O31", "O32", "O33", "O34", "O35", "O36", "O37", "O38", "O39", "O40",
                   "O41", "O42", "O43", "O44", "O45", "O46", "O47", "O48", "O49",
                   "V1", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10",
                   "V11", "V12", "V13", "V14", "V15", "V16", "V17", "V18", "V19", "V20",
                   "V21", "V22", "V23", "V24", "V25", "V26", "V27", "V28", "V29", "V30",
                   "V31", "V32", "V33", "V34", "V35", "V36", "V37", "V38", "V39", "V40",
                   "V41", "V42", "V43", "V44", "V45", "V46", "V47", "V48", "V49")


pal_40 <- c("#1F78C8" ,"#ff0000", "#33a02c" ,"#6A33C2" ,"#ff7f00", 
            "#565656","#FFD700","#a6cee3", "#FB6496" ,"#b2df8a", 
            "#CAB2D6","#FDBF6F","#0000FF","#EEE685", "#C8308C", 
            "#FF83FA" ,"#C814FA", "#999999" , "#36648B", "#00E2E5",
            "#00FF00", "#778B00", "#BEBE00","#8B3B00","#A52A3C" ,
            "#F0A0FF" ,"#0075DC", "#993F00" ,"#4C005C","#191919", 
            "#005C31", "#2BCE48" ,"#FFCC99" ,"#808080", "#94FFB5",
            "#FFC300", "#C70039", "#900C3F", "#581845","#DAF7A6")


Omni_RelAbund40_plot <- Taxa.tab.meta %>%
  filter(Group == "Omnivore") %>%
  group_by(Sample, Genus)%>%
  mutate(RelAbund_G = sum(RelAbund)) %>%
  ungroup() %>%
  group_by(Group, Genus)%>%
  mutate(AvgRelAbund = mean(RelAbund_G)) %>%
  filter(AvgRelAbund >= .00403) %>%
  replace_na(list(Genus = 'Unclassified')) %>%
  ungroup() %>%
  group_by(Sample) %>%
  ggplot(aes(x=factor(Sample, level =H_level_order), y=RelAbund, fill = Genus)) +
  geom_bar(stat = "identity", color = "black", size = .1) +
  scale_fill_manual(values = pal_40) +
  scale_y_continuous(labels = scales::percent, limits = c(0,1)) +
  labs(x = "Sample", y = "Relative Abundance") +
  theme(axis.text.x=element_text(angle=55,hjust=1.1))

Veg_RelAbund40_plot <- Taxa.tab.meta %>%
  filter(Group == "Vegetarian") %>%
  group_by(Sample, Genus)%>%
  mutate(RelAbund_G = sum(RelAbund)) %>%
  ungroup() %>%
  group_by(Group, Genus)%>%
  mutate(AvgRelAbund = mean(RelAbund_G)) %>%
  filter(AvgRelAbund >= .0029) %>%
  ungroup() %>%
  group_by(Sample) %>%
  ggplot(aes(x=factor(Sample, level =H_level_order), y=RelAbund, fill = Genus)) +
  geom_bar(stat = "identity", color = "black", size = .1) +
  scale_fill_manual(values = pal_40) +
  scale_y_continuous(labels = scales::percent, limits = c(0,1)) +
  labs(x = "Sample", y = "Relative Abundance") +
  theme(axis.text.x=element_text(angle=55,hjust=1.1))

RelAbund_plot <- Taxa.tab.meta %>%
  filter(Group == "Omnivore" | Group == "Vegetarian") %>%
  group_by(Group, Sample, Genus)%>%
  mutate(RelAbund_G = sum(RelAbund)) %>%
  ungroup() %>%
  group_by(Group, Genus)%>%
  mutate(AvgRelAbund = mean(RelAbund_G)) %>%
  filter(AvgRelAbund >= 0.006) %>%
  replace_na(list(Genus = 'Unclassified')) %>%
  ungroup() %>%
  group_by(Sample) %>%
  ggplot(aes(x=factor(Sample, level =H_level_order), y=RelAbund, fill = Genus)) +
  geom_bar(stat = "identity", color = "black", size = .1) +
  scale_fill_manual(values = pal_40) +
  scale_y_continuous(labels = scales::percent, limits = c(0,1)) +
  labs(x = "Sample", y = "Relative Abundance") +
  theme(axis.text.x=element_text(angle=90,hjust=.75, vjust = .5, size = 6)) +
  facet_wrap(Group~., scales = "free_x")


print(RelAbund_plot)
print(Omni_RelAbund40_plot)
print(Veg_RelAbund40_plot)

ggsave(paste0(WD, "/dada2_output/H_RelAbund_plot.png"), RelAbund_plot, width = 15, height = 8, units = "in") 
ggsave(paste0(WD, "/dada2_output/Omni_RelAbund40_plot.png"), Omni_RelAbund40_plot, width = 11.5, height = 8, units = "in") 
ggsave(paste0(WD, "/dada2_output/Veg_RelAbund40_plot.png"), Veg_RelAbund40_plot, width = 11.5, height = 8, units = "in") 
```

```{r final_save}
save.image(paste0(WD, "/dada2_output/homogeneity-workspace.RData"))
```

